pkg/apis/core/validation/validation.go

```diff
@@ -2129,9 +2129,6 @@ func validateDataSourceRef(dataSourceRef *core.TypedObjectReference, fldPath *fi
 	}
 
 	if len(dataSourceRef.Namespace) > 0 {
-		if dataSourceRef.Kind != "VolumeSnapshot" {
-			allErrs = append(allErrs, field.Invalid(fldPath, dataSourceRef.Kind, "expected value is VolumeSnapshot"))
-		}
 		if dataSourceRef.APIGroup == nil || *dataSourceRef.APIGroup != "snapshot.storage.k8s.io" {
```

Comment:
I think we should also remove the check on snapshot API group too if we want to allow any data populator.

Reply:
I see. Removed the check on snapshot API group